### DISCUSSION TITLE
[RFC] Looking to implement support for curl options in the storage plugin

### DISCUSSION THREAD

**Discussion by lnikkila** [MusicPlayerDaemon/MPD#1987] [Ideas]
Heyo!

I've been playing with MPD's WebDAV support for a while, and I realised the curl input plugin supports configuration options, but the storage plugin doesn't use those same options yet:
https://github.com/MusicPlayerDaemon/MPD/blob/c71e586c530e1d066efde7c2ca40c363f74100c7/src/storage/plugins/CurlStorage.cxx#L92

This means that if you're e.g. using HTTPS and configure `cafile`, playback works fine, but updating the database fails with:
```
mpd: exception: CURL failed: server certificate verification failed. CAfile: none CRLfile: none
```

While I'm not the greatest at C++, this seems like something I could work on, and I'm looking for feedback from someone with more experience with the codebase before diving deeper.

The way I see it, there are three ways to implement this:

1. Somehow reach my grubby little hands into the input plugin configuration, which is initialised into [these variables](https://github.com/MusicPlayerDaemon/MPD/blob/c71e586c530e1d066efde7c2ca40c363f74100c7/src/input/plugins/CurlInputPlugin.cxx#L132). This feels a little icky, but could be relatively simple.
2. Duplicate the same options in the storage plugin code and read the input plugin configuration separately there. This duplicates a little code, but would be clean.
3. Extract the curl configuration from the input plugin to something reusable that both plugins could use. This feels the most complex to me, since the lifecycles of these two plugins are completely different.

Thoughts? Other approaches I've missed?

**Comment by lnikkila** [comment]
@MaxKellermann Sorry to tag you directly, but I was wondering if you had any thoughts around this :)

**Reply by jcorporation** [reply]
I am not Max, but I would prefer a completely new set of options.

**Reply by lnikkila** [reply]
Thanks! I think that's the approach I'll investigate then
